org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/servicereference/AbstractServiceReferenceAssert.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.servicereference;
+
+import static org.osgi.test.assertj.bundle.BundleAssert.BUNDLE;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.assertj.core.api.AbstractComparableAssert;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.assertj.core.api.InstanceOfAssertFactory;
+import org.assertj.core.api.ListAssert;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.test.assertj.bundle.BundleAssert;
+import org.osgi.test.assertj.dictionary.DictionaryAssert;
+
+public abstract class AbstractServiceReferenceAssert<SELF extends AbstractServiceReferenceAssert<SELF, ACTUAL>, ACTUAL extends ServiceReference<?>>
+	extends AbstractComparableAssert<SELF, ACTUAL> {
+
+	protected AbstractServiceReferenceAssert(ACTUAL actual, Class<?> selfType) {
+		super(actual, selfType);
+	}
+
+	public DictionaryAssert<String, Object> hasServicePropertiesThat() {
+		return new DictionaryAssert<String, Object>(actual.getProperties());
+	}
+
+	public BundleAssert hasBundleThat() {
+		return isNotNull().extracting(actual -> actual.getBundle(), BUNDLE)
+			.as(actual + ".bundle");
+	}
+
+	@SuppressWarnings("rawtypes")
+	static InstanceOfAssertFactory<List, ListAssert<Bundle>> BUNDLE_LIST = InstanceOfAssertFactories.list(Bundle.class);
+
+	public ListAssert<Bundle> hasUsingBundlesThat() {
+		return isNotNull().extracting(bundleContext -> Arrays.asList(bundleContext.getUsingBundles()), BUNDLE_LIST)
+			.as(actual + ".usingBundles");
+	}
+
+	public SELF isRegisteredInBundle(Bundle bundle) {
+
+		isNotNull();
+		BundleAssert.assertThat(actual.getBundle())
+			.as("isRegisteredInBundle %s:%s", bundle)
+			.isEqualTo(bundle);
+		return myself;
+	}
+
+	public SELF isRegisteredInBundle(String bundleSymbolicName) {
+
+		isNotNull();
+		BundleAssert.assertThat(actual.getBundle())
+			.as("isRegisteredInBundle %s", bundleSymbolicName)
+			.hasSymbolicName(bundleSymbolicName);
```

Comment:
For assertions I find it's usually better not to delegate to another assertion syntax using the "as" formula. You can produce more fluent error messages by calling `failure()` directly (as you've done in with `isAssignableTo()`).